### DISCUSSION TITLE
Why is the `pending` state always false when using `useFormStatus` in a React form?

### DISCUSSION THREAD

**Discussion by theishu4** [vercel/next.js#72419] [Help]
### Summary

I'm using the `useFormStatus` hook from `react-dom` to track the pending state in my React form submission, but the `pending` state is always `false`, even when the form is being submitted. I want to disable the form fields and show the loading state when the form is submitting. 

Can anyone explain why the `pending` state doesn't reflect the form submission process, and how I can correctly manage the form's submission state?

---
`FormFields Component`
```tsx
const FormFields = ({
  form,
  formReturnMsg,
}: {
  form: UseFormReturn<SignupFormType>;
}) => {
  const { pending } = useFormStatus(); // pending always returns false

  return (
    <>
      <FormField
        control={form.control}
        name="email"
        render={({ field }) => (
          <Input
            placeholder="Email"
            type="email"
            disabled={pending}  // Should be disabled during submission
            {...field}
          />
        )}
      />
      <Button type="submit" disabled={pending}>
        {pending ? "Signing up..." : "Signup"}  // Button text should change on submit
      </Button>
    </>
  );
};;
```

---
`SignupForm Component`

```tsx
const SignupForm = () => {
  const [formReturnMsg, setFormReturnMsg] = useState<FormReturnType | null>(
    null,
  );
  const form = useForm<SignupFormType>({
    resolver: zodResolver(signupFormSchema),
  });

  const onSubmit = (values: SignupFormType) => {
    signupAction(values).then((value) => {
      setFormReturnMsg(value);
    });
  };

  return (
    <CardWrapper type="signup">
      <Form {...form}>
        <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-4">
          <FormFields form={form} formReturnMsg={formReturnMsg} />
        </form>
      </Form>
    </CardWrapper>
  );
};
```

### **What I've tried:**
- I've tried using `useFormStatus` to track the form submission state (`pending`) but it always returns `false`, even during form submission.

### **What I want to achieve:**
- I want the `pending` state to reflect the form's submission process, disabling the form fields and the submit button during submission.

**Comment by icyJoseph** [comment]
The action has to be passed as a `<form action={handler}>` kind of thing ~ not as the onSubmit prop...
